src/schema/sale_artwork.ts

```diff
@@ -84,7 +86,18 @@ export const SaleArtworkType = new GraphQLObjectType<any, ResolverContext>({
     return {
       ...GravityIDFields,
       cached,
-      artwork: { type: Artwork.type, resolve: ({ artwork }) => artwork },
+      artwork: {
+        type: Artwork.type,
+        resolve: ({ artwork }, _options, { artworkLoader }, info) => {
+          if (!info.fieldNodes) return artwork
+          const fetchArtworkFields = ["edition_of", "edition_sets"]
+          if (hasIntersectionWithSelectionSet(info, fetchArtworkFields)) {
+            return artworkLoader(artwork.id).catch(() => artwork)
```

Comment:
What’s the catch for? That would only break later on when trying to pluck fields from the artwork that aren’t there, right? By that time the original error is gone, so I think it would be better to error here.

Reply:
the catch was a safe back up plan in case there was an issue with fetching the artwork we basically fallback to artwork returned from sale artwork.

Reply:
Ok, let’s not do that and let it fail with the helpful error when it happens.